anomali-limo-feed/operations.py

```diff
@@ -0,0 +1,217 @@
+                        ],
+                        "modified": "",
+                        "name": "",
+                        "object_marking_refs": [
+                        ],
+                        "pattern": "",
+                        "type": "",
+                        "valid_from": ""
+                    }
+                ],
+                "spec_version": "",
+                "type": ""
+            })
+    except Exception as e:
+        logger.exception('An exception occurred {}'.format(e))
+        raise ConnectorError('An exception occurred {}'.format(e))
+
+
+def get_api_root_information(config, params):
+    try:
+        taxii = TaxiiClient(config)
+        params = get_params(params)
+        return taxii.make_request_taxii(params=params)
+    except Exception as e:
+        logger.exception('An exception occurred {}'.format(e))
+        raise ConnectorError('An exception occurred {}'.format(e))
+
+
+def get_collections(config, params):
+    try:
+        taxii = TaxiiClient(config)
+        params = get_params(params)
+        if params:
+            response = taxii.make_request_taxii(endpoint='collections/' + params['collectionID'] + '/')
+        else:
+            response = taxii.make_request_taxii(endpoint='collections/')
+        if response.get('collections'):
+            return response
+        else:
+            return {'collections': [response]}
+    except Exception as e:
+        logger.exception('An exception occurred {}'.format(e))
+        raise ConnectorError('An exception occurred {}'.format(e))
+
+
+def get_objects_by_collection_id(config, params):
+    try:
+        taxii = TaxiiClient(config)
+        if params.get('limit') is None or params.get('limit') == '':
+            server_url = config.get('server_url')
+            if not server_url.startswith('https://'):
+                server_url = 'https://' + server_url
+            if not server_url.endswith('/'):
+                server_url += '/'
+            username = config.get('username')
+            password = config.get('password')
+            collection = Collection(
+                server_url + 'api/v1/taxii2/feeds/collections/' + str(params.get('collectionID')) + '/',
+                user=username, password=password)
+            response = []
+            for bundle in as_pages(collection.get_objects, added_after=params.get('added_after') if (
+                    params.get('added_after') is not None and params.get(
+                'added_after') != '') else '1970-01-01T00:00:00.000Z', start=params.get('offset'),
+                                   per_request=1000):
+                response.append(bundle)
+        else:
+            params = get_params(params)
+            wanted_keys = set(['added_after'])
+            query_params = {k: params[k] for k in params.keys() & wanted_keys}
+            headers = {'Range': 'items {0}-{1}'.format(str(params.get('offset')), str(params.get('limit') - 1))}
```

Comment:
does this work correctly if a limit value of more than 1000 is specified? I have seen issues with some endpoints and they return only 1000 feeds

Reply:
yes mam it works if we specified limit value more than 1000